From b86cb74443a9e7328b9eea29dff86209414d83cf Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Sat, 26 Nov 2022 08:22:32 +0530
Subject: [PATCH] Fix build error in few setups due to change introduced for
 clang-13 bug

Revert "Kconfig: Add option for asm goto w/ tied outputs to workaround clang-13 bug"

This reverts commit 64639d11186e208f371cbbff74115514ac53a606 as the fix
is intended as workaround for clang-13 bug whereas in Android 13 clang-14
is used.

Tracked-On: OAM-105014
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 init/Kconfig | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/init/Kconfig b/init/Kconfig
index 641cd05d3e6b..722aa998006e 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -77,11 +77,6 @@ config CC_HAS_ASM_GOTO_OUTPUT
 	depends on CC_HAS_ASM_GOTO
 	def_bool $(success,echo 'int foo(int x) { asm goto ("": "=r"(x) ::: bar); return x; bar: return 0; }' | $(CC) -x c - -c -o /dev/null)
 
-config CC_HAS_ASM_GOTO_TIED_OUTPUT
-	depends on CC_HAS_ASM_GOTO_OUTPUT
-	# Detect buggy gcc and clang, fixed in gcc-11 clang-14.
-	def_bool $(success,echo 'int foo(int *x) { asm goto (".long (%l[bar]) - .\n": "+m"(*x) ::: bar); return *x; bar: return 0; }' | $CC -x c - -c -o /dev/null)
-
 config TOOLS_SUPPORT_RELR
 	def_bool $(success,env "CC=$(CC)" "LD=$(LD)" "NM=$(NM)" "OBJCOPY=$(OBJCOPY)" $(srctree)/scripts/tools-support-relr.sh)
 
-- 
2.38.1

